nodejs/index.js

```diff
@@ -1,57 +1,61 @@
-const { DeltaChat, C } = require('deltachat-node')
-const path = require('path')
+const { DeltaChat, C } = require("deltachat-node");
+const path = require("path");
 
 // Load config
-var conf = require('rc')("dc-echo", {
-    email_address: undefined,
-    email_password: undefined,
-})
+var conf = require("rc")("dc-echo", {
+  email_address: undefined,
+  email_password: undefined
+});
 
 // Setup DC
-const dc = new DeltaChat()
+const dc = new DeltaChat();
 
 function handleDCMessage(chatid, msgId) {
-    const chat = dc.getChat(chatid)
-    const msg = dc.getMessage(msgId)
+  const chat = dc.getChat(chatid);
+  const msg = dc.getMessage(msgId);
 
-    // only echo to DM
-    if (chat.getType() === C.DC_CHAT_TYPE_SINGLE) {
-        dc.sendMessage(chatid, msg.getText())
-    }
+  // only echo to DM
+  if (chat.getType() === C.DC_CHAT_TYPE_SINGLE) {
+    dc.sendMessage(chatid, msg.getText());
+  }
 }
 
-dc.on('DC_EVENT_MSGS_CHANGED', (chatId, msgId) => {
-    // Deaddrop fix for bot, otherwise first message would be ignored
-    const message = dc.getMessage(msgId)
-    if (message && message.isDeadDrop()) {
-        handleDCMessage(dc.createChatByMessageId(msgId), msgId)
-    }
-})
-dc.on('DC_EVENT_INCOMING_MSG', handleDCMessage)
+dc.on("DC_EVENT_MSGS_CHANGED", (chatId, msgId) => {
+  // Deaddrop fix for bot, otherwise first message would be ignored
+  const message = dc.getMessage(msgId);
+  if (message && message.isDeadDrop()) {
+    handleDCMessage(dc.createChatByMessageId(msgId), msgId);
+  }
+});
+dc.on("DC_EVENT_INCOMING_MSG", handleDCMessage);
 
-// dc.on('ALL', console.log.bind(null, 'core |')) // advanced logging for debugging
+//dc.on("ALL", console.log.bind(null, "core |")); // advanced logging for debugging
 
 // Start DC
-dc.open(path.join(__dirname, 'deltachat-db'), () => {
-    if (!dc.isConfigured()) {
-        if (!conf.email_address || !conf.email_password) {
-            console.error("Not configued and email address or password is missing.")
-            process.exit(1)
-        }        
-        dc.configure({
-            addr: conf.email_address,
-            mail_pw: conf.email_password,
-            e2ee_enabled: true,
-        }, () => {
-            console.log('init done')
-        })
-    } else {
-        console.log('init done')
+async function setup() {
+  await dc.open(path.join(__dirname, "deltachat-db"));
+  if (!dc.isConfigured()) {
+    if (!conf.email_address || !conf.email_password) {
+      console.error("Not configued and email address or password is missing.");
+      process.exit(1);
     }
-})
+    await dc.configure({
+      addr: conf.email_address,
+      mail_pw: conf.email_password,
+      e2ee_enabled: true,
+      // the bot flag is only respected core version >= 47, but it does not hurt to already add it
+      bot: true 
+    });
+  }
+  await dc.startIO();
+  console.log("init done");
+}
+
+setup();
```

Comment:
I don't have a setup to try this out right now, but don't we have to await this, as it is an async function? (I didn't write code in quite a while, if this is bullshit no need to explain in detail)

Reply:
nope, you cannot use await in the base/global/main/whatever-its-called scope in javascript.
That's why we need to wrap everything into an setup function. We could also use an [Immediately-invoked Function Expression](https://flaviocopes.com/javascript-iife/), but that concept might be harder to understand for js newbies:
```js
(async function () {
// content of setup funtion
})();
```